tests/ui/traits/winnowing/global-where-bound-region-constraints-2.rs

```diff
@@ -0,0 +1,31 @@
+//@ revisions: current next
+//@ ignore-compare-mode-next-solver (explicit revisions)
+//@[next] compile-flags: -Znext-solver
+//@ check-pass
+
+// Regression test for trait-system-refactor-initiative#172.
+//
+// In this test the global where-bound simply constrains the
+// object lifetime bound to 'static while the builtin impl
+// ends up also emitting a `dyn Any: 'static` type outlives
+// constraint. This previously resulted in ambiguity. We now
+// always prefer the impl.
+
+pub trait Any: 'static {}
+
+pub trait Downcast<T>: Any
+where
+    T: Any,
+{
+}
+
+impl dyn Any {
+    pub fn is<T>(&self)
+    where
+        T: Any,
+        Self: Downcast<T>,
```

Comment:
Please add a comment that elaborating this is where the global where bound comes from; it's not obviously clear from the test file.

Reply:
done